### PR TITLE
Clicking on tags displays the similarly tagged images.

### LINKED ISSUE #6: Images by Tag
#### As a user I want to view all images associated with a tag.

__Story__:

Now that we have added the ability for images to be associated with tags, let
us provide the ability for users to filter images by a given tag. Nana, who
loves fast planes, will be delighted to see only images labeled with SR-71.

__Acceptance criteria__:
- [x] When I click on a tag, I see a filtered list of only the images that have
  the tag I clicked on.
- [x] The tag name, not its ID, is used to perform the filtering.

__Discussion topics__:
- [ ] Where in the code should we implement this functionality?
- [ ] What happens if a user tries to manually filter by a nonexistent tag?

__Dependencies__:
- Image Tags

### PR DESCRIPTION
**Pull request by sagarayi**
Closes #6

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -3,4 +3,10 @@ def index
     # fetches all images present in the database
     @images = Image.all
   end
+
+  def filter
+    puts params[:selected_tag]
```
Comment (zju2) [inline app/controllers/home_controller.rb:8]: what about removing this print command?

File: test/controllers/home_controller_test.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 require 'test_helper'
 
 class HomeControllerTest < ActionDispatch::IntegrationTest
+  test 'should get filtered page' do
+    get home_filter_url, params: { selected_tag: 'photos' }
+    assert_response :success
+  end
```
Comment (zju2) [inline test/controllers/home_controller_test.rb:7]: what about adding a test where we have several images with several tags, when we click one tag, it returns correct images which have this tag?

File: app/controllers/home_controller.rb (hunk 1 above)
Comment (sagarayi) [reply app/controllers/home_controller.rb:8]: Yes, will remove it

File: test/controllers/home_controller_test.rb (hunk 2 above)
Comment (sagarayi) [reply test/controllers/home_controller_test.rb:7]: Sure, will add more test cases

**Pull request merged**